python/src/otel/otel_sdk/opentelemetry/instrumentation/aws_lambda/__init__.py

```diff
@@ -99,36 +96,32 @@ def _uninstrument(self, **kwargs):
 
 def _instrument(tracer, wrapped_module_name, wrapped_function_name):
     def _instrumented_lambda_handler_call(call_wrapped, instance, args, kwargs):
-        orig_handler_name = ".".join(
-            [wrapped_module_name, wrapped_function_name]
+        lambda_event, lambda_context = args[0], args[1]
+
+        parent_context = get_global_textmap().extract(
+            lambda_event.get("headers", {}).update(
+                {"X-Amzn-Trace-Id": environ.get("_X_AMZN_TRACE_ID", "")}
+            )
```

Comment:
How will this work if you have _both_ `TraceContextPropagator` and `AwsPropagator`? Will the context be a fix of both? With 2 spans?

Reply:
You should always use the xray propagator format when reading from the environment variable, and use the configured propagator if reading from HTTP headers when the environment variable wasn't sampled

Reply:
Hm I think I've seen in other Python code that they're able to do just `get_global_textmap().extract(headers)`. But we can't do that and have the behavior you mentioned if the user sets something like `OTEL_PROPAGATORS=xray,tracecontext` for their environment variables.

Maybe the answer is to read the `OTEL_PROPAGATORS` variable directly and instantiate the propagators? But that doesn't feel right... Would probably need to investigate the behavior of when we have multiple propagators.

Reply:
The env variable only ever fills in with AWS format so we can't really rely on configured propagator to read from it. In other languages we directly reference the awsxraypropagator through code without using the global can't you do that here?

Reply:
Yes that's what we intend to do here! We first try the `XRay` propagator, but if that fails, we try the global propagator. That's what I interpret the specs on [how to parent a Lambda trace](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/instrumentation/aws-lambda.md#determining-the-parent-of-a-span) saying we should do?

After updating the PR it looks like this:

```python
parent_context = None

xray_env_var = os.environ.get("_X_AMZN_TRACE_ID")

if xray_env_var:
    parent_context = AwsXRayFormat().extract(
        {"X-Amzn-Trace-Id": xray_env_var}
    )

event_span_attributes = {}

if not parent_context or not parent_context.trace_flags.sampled:
    lambda_event = args[0]
    # ... Try the global propagator
```